packages/headless/src/features/commerce/interactive-result/interactive-result-selectors.ts

```diff
@@ -0,0 +1,58 @@
+import {Ec, Product} from '@coveo/relay-event-types';
+import {createSelector} from '@reduxjs/toolkit';
+import {ProductRecommendation} from '../../../api/search/search/product-recommendation';
+import {CommerceEngineState} from '../../../app/commerce-engine/commerce-engine';
+import {getCurrency} from '../context/context-selector';
+import {ProductListingV2State} from '../product-listing/product-listing-state';
+import {CommerceSearchState} from '../search/search-state';
+
+export const getECProductClickPayload = (
+  product: ProductRecommendation,
+  state: CommerceEngineState
+): Ec.ProductClick => ({
+  product: formatProduct(product),
+  responseId: getResponseId({...state.productListing, ...state.commerceSearch}),
```

Comment:
When the RFC around multiple solution types resolves, we should have a way to create an interactive result controller from its parent controller. This means we'll only have access to the state from the solution type in question, which will make finding the `responseId` easy-peasy.

However, if we want a single interactive result controller that can be created without a reference to its parent, I don't think this approach will work. For example, let's take a listing page with a recs carroussel as example:
1. User visits page
2. Listing is received with `responseId` = `listing-response-id`
3. Recs are received with `responseId` = `recs-response-id`
4. User clicks on listing product, which would use `recs-response-id` 🔴

If we really want a single controller, we have to put the "control" in the hands of the implementer:
1. We let them use the state themselves ([as you proposed!](https://coveord.atlassian.net/browse/LENS-1595))
2. Or, we create multiple controllers

Reply:
Let's put this jira on hold until the RFC is ready? Is there a jira we can link to in the CAPI project?

Reply:
Yes! https://coveord.atlassian.net/browse/CAPI-496

Reply:
Thanks for your inputs! I will close the PR for now, and will revisit after the RFC is approved!